### PR TITLE
[#216] [#202] Enable Dex SSO on jenkins

### PR DESCRIPTION
**Pull request by MaksymBilenko**
This PR fixes issues #216 #202

### CODE REVIEW

**Comment by MaksymBilenko** [comment]
test this please

**Comment by aliaksandr-d** [comment]
Test this please

**Comment by aliaksandr-d** [comment]
test this please

**Comment by MaksymBilenko** [comment]
test this please

**Comment by aliaksandr-d** [comment]
test this please

**Comment by aliaksandr-d** [comment]
test this please

**Comment by MaksymBilenko** [comment]
test this please

**Comment by aliaksandr-d** [comment]
test this please

**Comment by aliaksandr-d** [comment]
test this please

**Comment by MaksymBilenko** [comment]
test this please

**Comment by kirillmakhonin** [comment]
Test this please

File: k8s/jenkins/scripts/security.groovy [hunk 1]
```diff
@@ -35,4 +106,63 @@ if (hudsonRealm.getAllUsers().size == 0){
 	Jenkins.instance.getInjector().getInstance(AdminWhitelistRule.class).setMasterKillSwitch(false)
 }
 
-System.setProperty("hudson.model.DirectoryBrowserSupport.CSP", "")
\ No newline at end of file
+enableOpenId = env['OPENID_ENABLE']
+if (enableOpenId.toLowerCase() == "true") {
+  String clientId = env['OPENID_CLIENTID']
```
Comment (kirillmakhonin) [inline k8s/jenkins/scripts/security.groovy]: @MaksymBilenko  As I see, any job on Jenkins can access this env. variables with critical information. Am I right?

**Comment by kirillmakhonin** [comment]
test this please

**Comment by MaksymBilenko** [comment]
test this please

File: k8s/jenkins/scripts/security.groovy (hunk 1 above)
Comment (MaksymBilenko) [reply k8s/jenkins/scripts/security.groovy]: All Jenkins jobs are running on slaves where we don't have this variables.

**Comment by MaksymBilenko** [comment]
test this please

**Comment by ablatov** [comment]
test this please

**Comment by MaksymBilenko** [comment]
test this please

**Comment by ablatov** [comment]
test this please

**Comment by ablatov** [comment]
test this please

**Comment by aliaksandr-d** [comment]
Test this please

**Comment by ablatov** [comment]
Test this please

**Comment by aliaksandr-d** [comment]
test this please

File: legion_test/bin/dex_client [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+def work():
+    profiler_loader.get_variables(os.getenv('PATH_TO_PROFILES_DIR'))
+    if dex_client.get_jenkins_credentials():
+        username, password = dex_client.get_jenkins_credentials()
+        cookies = ';'.join(['{}={}'.format(k, v) for (k, v) in dex_client.get_session_cookies().items()])
+        print('{}\n{}\n{}\n'.format(username, password, cookies))
+
+
+if __name__ == '__main__':
+    parser = argparse.ArgumentParser('Dex client')
```
Comment (kirillmakhonin) [inline legion_test/bin/dex_client]: @aliaksandr-d maybe we should rename this tool to make it clear that it generates only Jenkins credentials (like jenkins-dex-client)?

**Comment by aliaksandr-d** [comment]
@kirillmakhonin fixed. please review.

**Comment by aliaksandr-d** [comment]
test this please

**Comment by aliaksandr-d** [comment]
test this please

**Comment by aliaksandr-d** [comment]
Test this please

**Comment by aliaksandr-d** [comment]
Test this please

**Pull request merged**